src/main/resources/answer/acceptAnswerById.xml

```diff
@@ -0,0 +1,86 @@
+  ~ under the License.
+  -->
+<template xmlns="http://ws.apache.org/ns/synapse" name="acceptAnswerById">
+    <parameter name="id" description="Id of the answer"/>
+    <!-- OPTIONAL -->
+    <parameter name="filter" description="Response filter"/>
+    <parameter name="preview"
+               description="Preview parameter is used for development and testing purposes. When preview is set to true,
+                            request will only simulate whether it is a valid request or not and will not change
+                            anything on the site."/>
+    <sequence>
+        <property name="uri.var.id" expression="$func:id"/>
+        <!-- OPTIONAL -->
+        <property name="uri.var.filter" expression="$func:filter"/>
+        <property name="uri.var.preview" expression="$func:preview"/>
+
+        <payloadFactory media-type="json">
+            <format>
+                {
+                "site": "$1",
+                "access_token": "$2",
+                "key": "$3"
+                }
+            </format>
+            <args>
+                <arg evaluator="xml" expression="$ctx:uri.var.site"/>
+                <arg evaluator="xml" expression="$ctx:uri.var.accessToken"/>
+                <arg evaluator="xml" expression="$ctx:uri.var.key"/>
+            </args>
+        </payloadFactory>
+        <script language="js">
+            <![CDATA[
+                var payload = mc.getPayloadJSON();
+
+                function isOptionalEmpty(optional) {
+                    if (typeof optional === 'undefined') {
+                        return true;
+                    }
+                    if (optional !== null && optional !== '') {
+                        return false;
+                    }
+                    return true;
+                }
+
+                function setNonNullOptional(payload, optionalName) {
+                    var optional = mc.getProperty('uri.var.' + optionalName);
+                    if (!isOptionalEmpty(optional)) {
+                        payload[optionalName] = String(optional);
+                    }
+                }
+
+                setNonNullOptional(payload, 'preview');
+                setNonNullOptional(payload, 'filter');
+
+                mc.setPayloadJSON(payload);
+            ]]>
+        </script>
+        <!-- Sets header before conversion -->
+        <header name="Content-Type" scope="transport" value="application/x-www-form-urlencoded"/>
+        <!-- Converts request -->
+        <property name="messageType" value="application/x-www-form-urlencoded" scope="axis2" type="STRING"/>
+        <call>
+            <endpoint>
+                <http method="POST"
+                      uri-template="{uri.var.apiDomain}/{uri.var.apiVersion}/answers/{uri.var.id}/accept"/>
+            </endpoint>
+        </call>
+        <!-- Forces response decompression -->
+        <class name="org.wso2.carbon.connector.DecompressMediator"/>
+    </sequence>
+</template>
```

Comment:
Add new line at the end and fix in all places

Reply:
Fix with 0f0b739